tests/virt_autotest/sev_snp_guest_verification.pm

```diff
@@ -0,0 +1,1270 @@
+sub configure_sev_snp_kernel_parameters {
+    my $self = shift;
+
+    # Required kernel parameters for SEV-SNP
+    my %required_params = (
+        'amd_iommu=on' => 'AMD IOMMU support',
+        'iommu=nopt' => 'IOMMU no pass-through mode',
+        'mem_encrypt=on' => 'Memory encryption',
+        'kvm_amd.sev=1' => 'SEV base functionality',
+        'kvm-amd.sev_snp=1' => 'SEV-SNP functionality'
+    );
+
+    # Check current kernel parameters
+    my $cmdline = script_output("cat /proc/cmdline");
+    record_info('Kernel cmdline', $cmdline);
+
+    # Find missing parameters
+    my @missing;
+    foreach my $param (keys %required_params) {
+        push @missing, $param unless $cmdline =~ /\b$param\b/;
+    }
+
+    # If all parameters are present, return success
+    if (!@missing) {
+        record_info('SEV-SNP boot parameters', 'All required parameters present');
+        return 1;
+    }
+
+    # Log missing parameters
+    my $missing_list = join(", ", @missing);
+    record_info('Missing SEV-SNP parameters', "Missing parameters: $missing_list", result => 'softfail');
+
+    # Add missing parameters using bootloader_setup library
+    record_info('Fixing GRUB', "Adding missing SEV-SNP parameters to GRUB configuration");
+    foreach my $param (@missing) {
+        add_grub_cmdline_settings($param);
+    }
+
+    # Update GRUB configuration
+    grub_mkconfig();
+    record_info('GRUB updated', 'GRUB configuration updated with missing parameters');
+
+    # Reboot to apply changes
+    record_info('Reboot required', 'Rebooting to apply new parameters');
+    power_action('reboot', textmode => 1);
+
+    # Wait for boot completion - robust parameters for systems with/without GRUB menu
+    $self->wait_boot(textmode => 1, bootloader_time => 60, ready_time => 300);
+    select_serial_terminal;
+
+    # Verify parameters after reboot
+    my $new_cmdline = script_output("cat /proc/cmdline");
+    record_info('New kernel cmdline', $new_cmdline);
+
+    # Check if all parameters are now present
+    my $all_present = 1;
+    foreach my $param (keys %required_params) {
+        if ($new_cmdline !~ /\b$param\b/) {
+            record_info("Parameter missing", "Required parameter '$param' (${required_params{$param}}) not applied", result => 'fail');
+            $all_present = 0;
+        }
+    }
+
+    if (!$all_present) {
+        die "SEV-SNP verification requires specific kernel parameters. Not all parameters were applied after reboot. Test cannot continue.";
+    }
+
+    record_info('Parameters Applied', "All SEV-SNP parameters successfully applied after reboot");
+    return 1;
+}
+
+=head2 setup_sles15sp6_coco_packages
+
+  setup_sles15sp6_coco_packages($self)
+
+Install and configure Confidential Computing (COCO) packages for SLES 15-SP6.
+This includes installing kernel-coco, QEMU, libvirt, and OVMF packages,
+locking them to prevent updates, and rebooting to activate the COCO kernel.
+
+=cut
+
+sub setup_sles15sp6_coco_packages {
```

Comment:
On sle15sp7 host, sle15sp6 and sle15sp7 guests are both valid to be tested. For sle15sp6 guest, it will need to setup pkgs with coco module, eg kernel-default-coco, etc. I do not find this part. Would you please point me?

Reply:
it is for sles15sp6 host sev-snp, it is TP version. 
Removed.